Tools/sitl_run.sh

```diff
@@ -100,6 +100,12 @@ done
 
 export PX4_SIM_MODEL=${model}
 
+if [[ "$sitl_bin" == *"px4_sitl_sih"* ]]; then
+	export PX4_SIH="true"
+else
+	export PX4_SIH="false"
+fi
+
```

Comment:
Could we try to use the `$program` so that we can prevent being able to run SIH with e.g. gazebo? or was this intentional that this was done independently?

Reply:
I'm just very new to these starting scripts. So I'm trying to make something functional to start with :)
Currently, I'm compiling and launching as 
`make px4_sitl_sih none`
I guess you'd like to see
`make px4_sitl sih`? 

Sih is not a viewer, this is a simulation engine. Technically speaking we could run the sih as simulation engine and jmavsim as the viewer. The command could be something like `make px4_sitl_sih jmavsim`.
Knowing that. Does that make sense as it is?

Reply:
I guess we could do
`make px4_sitl sih` to run the sih without a viewer, in this case only QGC can display the vehicle motion.
`make px4_sitl sih___jmavsim` in that case jmavsim is the 'world' and we start it as display only.

Reply:
> I guess we could do
make px4_sitl sih to run the sih without a viewer, in this case only QGC can display the vehicle motion.
make px4_sitl sih___jmavsim in that case jmavsim is the 'world' and we start it as display only.

I don't think we should do this is the right structure. 

While `$viewer` option is analogous to a simulation engine, and SITL does not distinguish these two. Usually if a separate viewer exists, this is turned off with the `HEADLESS=1` option. Therefore I think it makes sense to consider SIH as the same level of `gazebo` / `jmavsim` etc

A "world" is not what kind of viewer/simulation engine you are using but a location/environment you are flying in. Currently jsbsim / gazebo uses the concept of world this way so I don't think we should introduce a new structure for SIH

> make px4_sitl sih to run the sih without a viewer, in this case only QGC can display the vehicle motion.

For me it would make more sense to do `make px4_sitl sih` that runs the viewer by default. If you want to turn off the viewer, to run `HEADLESS=1 make px4_sitl sih` since this makes it consistent with the other simulation startup targets

Reply:
If this is too strange, I am open for suggestions but I would prefer making all SITL targets at least consistent so it is not confusing to move between different simulation engines

Reply:
ok, here we can run the sih as `HEADLESS=1 make px4_sitl_nolockstep sih`
Without the headless, jmavsim starts but is not connected to PX4 since the module `simulator` is not started.

TODO
1. connect the sih to jmavsim for its display
2. select the airframe. The sih can simulate a quadrotor or an airplane